src/infra/core/src/use_cases/commit_dataset_event_use_case_impl.rs

```diff
@@ -10,37 +10,76 @@
 use std::sync::Arc;
 
 use dill::{component, interface};
-use kamu_core::auth::{DatasetAction, DatasetActionAuthorizer, DatasetActionUnauthorizedError};
+use internal_error::ErrorIntoInternal;
+use kamu_core::auth::DatasetActionUnauthorizedError;
 use kamu_core::{
     CommitDatasetEventUseCase,
     DatasetLifecycleMessage,
     DatasetRegistry,
+    EditDatasetUseCase,
+    EditDatasetUseCaseError,
+    NotAccessibleError,
+    ViewDatasetUseCase,
     MESSAGE_PRODUCER_KAMU_CORE_DATASET_SERVICE,
 };
 use messaging_outbox::{Outbox, OutboxExt};
+use odf::dataset::{AppendError, InvalidEventError};
+use odf::metadata::{EnumWithVariants, TransformInputExt};
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
 #[component(pub)]
 #[interface(dyn CommitDatasetEventUseCase)]
 pub struct CommitDatasetEventUseCaseImpl {
     dataset_registry: Arc<dyn DatasetRegistry>,
-    dataset_action_authorizer: Arc<dyn DatasetActionAuthorizer>,
+    view_dataset_use_case: Arc<dyn ViewDatasetUseCase>,
+    edit_dataset_use_case: Arc<dyn EditDatasetUseCase>,
     outbox: Arc<dyn Outbox>,
 }
 
 impl CommitDatasetEventUseCaseImpl {
     pub fn new(
         dataset_registry: Arc<dyn DatasetRegistry>,
-        dataset_action_authorizer: Arc<dyn DatasetActionAuthorizer>,
+        view_dataset_use_case: Arc<dyn ViewDatasetUseCase>,
+        edit_dataset_use_case: Arc<dyn EditDatasetUseCase>,
         outbox: Arc<dyn Outbox>,
     ) -> Self {
         Self {
             dataset_registry,
-            dataset_action_authorizer,
+            view_dataset_use_case,
+            edit_dataset_use_case,
             outbox,
         }
     }
+
+    async fn validate_event(
+        &self,
+        event: odf::MetadataEvent,
+    ) -> Result<odf::MetadataEvent, odf::dataset::CommitError> {
+        if let Some(set_transform) = event.as_variant::<odf::metadata::SetTransform>() {
+            let mut inputs_dataset_refs = Vec::with_capacity(set_transform.inputs.len());
+            for input in &set_transform.inputs {
+                let input_dataset_ref = input.as_sanitized_dataset_ref()?;
+                inputs_dataset_refs.push(input_dataset_ref);
+            }
+
+            let view_result = self
+                .view_dataset_use_case
+                .execute_multi(inputs_dataset_refs)
+                .await?;
```

Comment:
Rather than a use case invoking another use case, it could have been invoking a shared helper.

Reply:
Discussed in Slack

---

Shared helper (`AccessDatasetHelper`) has been extracted 

Done